csharp/src/Drivers/Apache/Hive2/HiveServer2Statement.cs

```diff
@@ -426,12 +427,128 @@ private async Task<QueryResult> GetQueryResult(TSparkDirectResults? directResult
                 int columnCount = HiveServer2Reader.GetColumnCount(rowSet);
                 int rowCount = HiveServer2Reader.GetRowCount(rowSet, columnCount);
                 IReadOnlyList<IArrowArray> data = HiveServer2Reader.GetArrowArrayData(rowSet, columnCount, schema, Connection.DataTypeConversion);
+
+                // Enhance column schema results if this is a GetColumns query
+                if (SqlQuery?.ToLowerInvariant() == GetColumnsCommandName)
```

Comment:
I think this looks clean, but I would prefer not to see a test for `GetColumnsCommandName` in something generic as `GetQueryResult`. But I can refactor this later. The functionality is correct. Thanks.